.github/workflows/build-container.yaml

```diff
@@ -1,68 +1,168 @@
+        uses: actions/checkout@v2
+      - id: setup-buildx
+        name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@b5ca514318bd6ebac0fb2aedd5d36ec1b5c232a2 #v3.10.0 
+      - id: container-login
+        if: ${{ env.PUBLISH }}
+        name: Log in to the Container registry
+        uses: docker/login-action@74a5d142397b4f367a81961eba4e8cd7edddf772 #v3.4.0
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - id: meta
+        if: ${{ env.PUBLISH }}
+        name: Extract metadata (tags, labels) for Docker
+        uses: docker/metadata-action@902fa8ec7d6ecbf8d84d538b9b233a880e428804 #v5.7.0
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+      - id: name-artifact
+        env:
+          REPO: ${{ github.repository }}
+          RELEASE: ${{ github.event_name == 'release' }}
+          TAG: ${{ steps.meta.outputs.tags || inputs.tag || env.TAG }}
+        run: |
+          if [ -z "$RELEASE" ]; then
+            echo "name=tag-release" >> "$GITHUB_OUTPUT"
+          else
+            tag=$(echo $TAG | sed -e "s+.*\:++")
+            echo "name=tag-$tag" >> "$GITHUB_OUTPUT"
+          fi
+        shell: bash {0}
+      - id: build
+        name: Build and export
+        uses: docker/build-push-action@471d1dc4e07e5cdedd4c2171150001c434f0b7a4 #v6.15.0
+        with:
+          context: .
+          file: ./Dockerfile
+          tags: ${{ steps.meta.outputs.tags || inputs.tag || env.TAG }}
+          outputs: type=docker,dest=/tmp/hub-dash-site-deployer.tar
+      - id: test
+        name: Test image
+        run: |
+          tag=${TAG/*:/}
+          docker load --input /tmp/hub-dash-site-deployer.tar
+          bash tests/run.sh "ghcr.io/hubverse-org/hub-dash-site-builder:${tag}" reichlab/flusight-dashboard
+      - id: upload
+        name: Upload artifact
+        uses: actions/upload-artifact@v4
+        with:
+          retention-days: 1
+          name: ${{ steps.name-artifact.outputs.name }}
+          path: /tmp/hub-dash-site-deployer.tar
+
+  test:
+    needs: [build-image]
+    if: ${{ github.event_name == 'pull_request' }}
+    name: "Test built image against published test suite (may fail)"
+    runs-on: ubuntu-latest
+    permissions: read-all
+    steps:
+      - id: check-artifacts
+        name: Fetch image
+        uses: actions/download-artifact@v4
+        with:
+          path: artifacts
+          pattern: tag*
+      - id: test-image
+        name: Test image against tests on main
+        run: |
+          tag=$(ls artifacts/)
+          path=$(ls artifacts/$tag/*)
+          docker load --input "$path"
+          tests=$(mktemp)
+          curl -o $tests https://raw.githubusercontent.com/hubverse-org/hub-dash-site-builder/refs/heads/main/tests/run.sh
+          bash ${tests} "ghcr.io/hubverse-org/hub-dash-site-builder:${tag}" reichlab/flusight-dashboard
```

Comment:
Fetching the test suite from the main branch and running it against the built image from a PR. 

If any of these tests fail, then it requires a deeper look into the process of the PR.

Reply:
Actually, that's a pretty good point to prevent a "the call is coming from inside the house" situation.

Reply:
done in 1693957f7fb123ea7cda63d3f09845471aadadd3